internal/servers/controller/common/scopeids/scope_ids.go

```diff
@@ -15,70 +16,127 @@ import (
 	"github.com/hashicorp/boundary/sdk/pbs/controller/api/resources/scopes"
 )
 
-// GetListingScopeIds, given common parameters for List calls, returns the set of scope
-// IDs in which to search for resources. It also returns a memoized map of the
-// scopes to their info for populating returned values.
-//
-// Note: This was originally pulled out 1:1 from the role service. It and other
-// tests in the other service handlers test this function extensively as it
-// forms the basis for all recursive listing tests; see those tests for list
-// functionality in the various service handlers.
-func GetListingScopeIds(
+type authzProtectedEntityProvider interface {
+	// Fetches basic resource info for the given scopes. Note that this is a
+	// "where clause" style of argument: if the set of scopes is populated these
+	// are the scopes to limit to (e.g. to put in a where clause). An empty set
+	// of scopes means to look in *all* scopes, not none!
+	FetchAuthzProtectedEntityInfo(context.Context, []string) (map[string][]boundary.AuthzProtectedEntity, error)
```

Comment:
having a named type for this map might help clarify what the keys need to be, maybe `AuthProtectedEntitiesByScope`? Figuring out a name for that map might also help with the function name, ie `FetchAuthzProtectedEntitiesByScope`

Also using named parameters for the interface might be a good idea, ie:
```
FetchAuthzProtectedEntityInfo(ctx context.Context, scopeIds []string)
```